design-docs/handling-encodings.md

```diff
@@ -0,0 +1,87 @@
+# Encoding
+
+Updated: __2016/03/19__
+Status: __Under Review__
+
+This document describes the rules and conventions for handling the character encoding of file contents in general, and file names 
+and comments in archives. It applies to tasks which need to read, write and transform text files, as well as tasks which create
+archives (zip files, tar files, etc.)
+
+## Use Cases
+
+### File Content Encoding
+
+Any time we need to generate a text file, or to read, transform and write the contents of a text file, we need to know what 
+character encoding(s) to use.
+
+Common situations include:
+
+* Copying a file from one location to another.
+* Creating an archive (zip, tar, etc.).
+* Transforming text in a file.
+
+[GRADLE-1267](https://issues.gradle.org/browse/GRADLE-1267) describes just one instance of where this functionality is needed.
+
+### File Name Encoding
```

Comment:
I'm not sure this is something that should actually be worried about. This is highly system dependent, while the content encoding is application dependent. AFAIK, java doesn't even provide a way to set the encoding of the filename. eg look at the constructor for File, it only takes a string. When converting to char* internally, it will use the system dependent `file.encoding`. To the filesystem, the filename is really just bytes. How those are interpreted depend on eg the encoding set in your terminal. This is why using ascii names is generally a good idea (as eg utf8 and win1252 are both ascii compatible).